File: leetcode/easy/1021. Remove Outermost Parentheses/solution.js
// https://leetcode.com/problems/remove-outermost-parentheses/
/**
 * @param {string} s
 * @return {string}
 */
function removeOuterParentheses(s) {
    let open = 0, close = 0;
    let res = '';

    for (const paren of s) {
        paren === '(' ? open++ : close++;
        if (open > 1 && open > close) {
            res += paren;
        }
        if (open === close) {
            open = close = 0;
        }
    }

    console.log(res)
    return res;
};

removeOuterParentheses(`(()())(())`)
removeOuterParentheses(`(()())(())(()(()))`)

File: leetcode/easy/1021. Remove Outermost Parentheses/solution.cc
#include <string>

using namespace std;

class Solution {
public:
    string removeOuterParentheses(string s) {
        int count = 0;
        string res = "";
        for (char ch: s) {
            if (ch == '(') {
                if (count++) res += '(';
            }
            else {
                if (--count) res += ')';
            }
        }
        return res;
    }
};

int main() {
    return 0;
}